### PR TITLE
Bug-fix [Vhost] Resolve issue where APIs are not properly undeployed from specific host under specific environment.

### LINKED ISSUE #2448: Undeploy a revision from one env, when an API is deployed to two envs with same vhost
### Steps to reproduce:
- Setup two enforcer and router environments (US and Default) with one adapter [vhost-docker-compose-setup.tar.gz](https://github.com/wso2/product-microgateway/files/7570158/vhost-docker-compose-setup.tar.gz)
- Add a gateway from admin portal (US, with vhost, localhost, ports 9190, 9195)
- Deploy an API to the same vhost localhost, in Default and US environments
- Undeploy the API from Default
- Invoke the API in Default gateway
```
curl -X GET "https://localhost:9095/v2/1.0.5/pet/findByStatus?status=pending" -H "accept: application/json" -H "Authorization: Bearer $TOKEN"
```
```
{"error_message":"Resource forbidden ","code":"900908","error_description":"User is NOT authorized to access the Resource. API Subscription validation failed."}
```
- Config Dump from Router's admin portal
http://localhost:9000/config_dump
```
         {
          "match": {
           "headers": [
            {
             "name": ":method",
             "string_match": {
              "safe_regex": {
               "google_re2": {},
               "regex": "^(GET|OPTIONS)$"
              }
             }
            }
           ],
           "safe_regex": {
            "google_re2": {},
            "regex": "^/v2/1.0.5/pet/findByStatus(\\?([^/]+))?$"
           }
          },
```

http://localhost:9100/config_dump
```
         {
          "match": {
           "headers": [
            {
             "name": ":method",
             "string_match": {
              "safe_regex": {
               "google_re2": {},
               "regex": "^(GET|OPTIONS)$"
              }
             }
            }
           ],
           "safe_regex": {
            "google_re2": {},
            "regex": "^/t/abc.com/v2/1.0.5/pet/findByStatus(\\?([^/]+))?$"
           }
          },
```

### Affected Product Version:
v1.0.0 alpha snapshot
v1.0.0 beta snapshot

### Environment details (with versions):
- OS: mac catelina
- Client: curl
- Env (Docker/K8s): docker

---

### LINKED ISSUE #2451: Undeploy an API from a vhost and in a specific environment using APICTL
### Description:
Undeploy an API from a vhost and in a specific environment using APICTL

### Steps to reproduce:
1. Setup two environment `Default` and `US` - [vhost-docker-compose-setup.tar.gz](https://github.com/wso2/product-microgateway/files/7571512/vhost-docker-compose-setup.tar.gz)
2. Deploy petstore API to two labels, with following deployment-enviornments file  [petstore.tar.gz](https://github.com/wso2/product-microgateway/files/7571527/petstore.tar.gz)

```yaml
type: deployment_environments
version: v4.0.0
data:
 -
   displayOnDevportal: true
   deploymentEnvironment: US
 -
   displayOnDevportal: true
   deploymentVhost: localhost
   deploymentEnvironment: US
 -
   displayOnDevportal: true
   deploymentEnvironment: Default

```
3. apictl mg deploy api -e dev -k -f petstore/
4. $ apictl mg get apis -e dev -k
Limit flag not set. Set to default: 25
APIs total: 2 received: 2
NAME                VERSION             TYPE                CONTEXT             GATEWAY_ENVS        VHOST
SwaggerPetstore     1.0.5               HTTP                /v2                 [US]                us.wso2.com
SwaggerPetstore     1.0.5               HTTP                /v2                 [US Default]        localhost
5. $ apictl mg undeploy api -n SwaggerPetstore -v 1.0.5 -e dev -k --vhost localhost -g US
API undeployed from microgateway successfully!
6. $ apictl mg get apis -e dev -k
Limit flag not set. Set to default: 25
APIs total: 2 received: 2
NAME                VERSION             TYPE                CONTEXT             GATEWAY_ENVS        VHOST
SwaggerPetstore     1.0.5               HTTP                /v2                 [US]                us.wso2.com
SwaggerPetstore     1.0.5               HTTP                /v2                 [Default]           localhost
7. $ TOKEN=$(curl -X POST "https://localhost:9095/testkey" -H "Authorization: Basic YWRtaW46YWRtaW4=" -k -v)
8. $ curl -X GET "https://localhost:9195/v2/store/inventory" -H "accept: application/json" -H "Authorization: Bearer $TOKEN" -k
{"test_status":6,"sold":20,"string":217,"bad":4,"sold ":1,"alive":1,"unavailable":1,"TESTEB":1,"pending":13,"available":662,"solt":1,"null":2,"connector":1,"extinct":3,"status":1}

Could invoke the undeployed API.

### Affected Product Version:
CC v1.0.0 beta-snapshot

### Environment details (with versions):
- OS: mac
- Client: curl
- Env (Docker/K8s): docker

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
$subject happens when the following conditions are met.

API is deployed in multiple gateway environments.
API deployment within one gateway environment has two vhosts.

When the API is deployed from specific host under specific gateway environment, the issue occurs.

refer to the issue for more information.

### Issues

Fixes #2451
FIxes #2448

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: integration/test-integration/src/test/resources/configs/multiple-env-config.toml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      authURLs = ["ssl://apim:9711"]
+
+[analytics]
+  enabled = false
+
+[enforcer.metrics]
+  enabled = false
+
+[enforcer.tracing]
+  enabled = false
```
Comment (pubudu538) [inline integration/test-integration/src/test/resources/configs/multiple-env-config.toml:90]: This is not used..

Comment (pubudu538) [reply integration/test-integration/src/test/resources/configs/multiple-env-config.toml:90]: Basically we removed this configuration.

**Review by renuka-fernando — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20211215.32

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211215.43

**Pull request merged**